core/src/core/transaction.rs

```diff
@@ -40,10 +40,12 @@ bitflags! {
 	/// Options for a kernel's structure or use
 	#[derive(Serialize, Deserialize)]
 	pub struct KernelFeatures: u8 {
-		/// No flags
-		const DEFAULT_KERNEL = 0b00000000;
-		/// Kernel matching a coinbase output
-		const COINBASE_KERNEL = 0b00000001;
+		/// plain kernel has fee, but no lock_height
+		const PLAIN         = 0;
+		/// coinbase kernel has neither fee nor lock_height (both zero)
+		const COINBASE      = 1;
+		/// absolute height locked kernel; has fee and lock_height
+		const HEIGHT_LOCKED = 2;
```

Comment:
I think this may be asking for trouble and its going to be easy to misinterpret this and add `3` as the next one, which will return true for both `is_coinbase()` and `is_height_locked()`.

Did we come to consensus on reworking these as enums and moving away from bitflags?

Reply:
These should definitely be recoded as enums. I used to not trust my rust skills to do it myself. But now that I encapsulated the getters with is_XXX() methods, I could give it a try.
You're right that 3 should be able to be used for something else entirely, like kernel-relative height locks.